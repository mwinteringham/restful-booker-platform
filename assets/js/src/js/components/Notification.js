import React, { useEffect } from 'react';
import { Link } from 'react-router-dom';

const Notification = ({setCount, count}) => {

    useEffect(() => {
        setCount();
    }, [])


    let alertIcon;

    if(count > 0){
        alertIcon = <i className="fa fa-inbox" style={{fontSize : "1.5rem"}}><span className="notification">{count}</span></i>
    } else {
        alertIcon = <i className="fa fa-inbox" style={{fontSize : "1.5rem"}}></i>
    }
    
    return <Link className="nav-link" to="/admin/messages">{alertIcon}</Link>;
    
}

export default Notification;
