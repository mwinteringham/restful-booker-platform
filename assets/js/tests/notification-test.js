import React from 'react';
import Notification from '../src/js/components/Notification.js';
import {
    render,
    asFragment,
  } from '@testing-library/react'
import { BrowserRouter } from 'react-router-dom';

test('Notification renders plain inbox when no unread notifications', () => {
    const {asFragment} = render(<BrowserRouter>
        <Notification setCount={() => {}} />
    </BrowserRouter>);

    expect(asFragment()).toMatchSnapshot();
});

test('Notification renders alert inbox when there are unread notifications', async () => {
    const {asFragment} = render(<BrowserRouter>
        <Notification setCount={() => {}} count={34} />
    </BrowserRouter>);

    expect(asFragment()).toMatchSnapshot();
});
