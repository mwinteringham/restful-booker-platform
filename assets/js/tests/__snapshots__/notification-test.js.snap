// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Notification renders alert inbox when there are unread notifications 1`] = `
<DocumentFragment>
  <a
    class="nav-link"
    href="/admin/messages"
  >
    <i
      class="fa fa-inbox"
      style="font-size: 1.5rem;"
    >
      <span
        class="notification"
      >
        34
      </span>
    </i>
  </a>
</DocumentFragment>
`;

exports[`Notification renders plain inbox when no unread notifications 1`] = `
<DocumentFragment>
  <a
    class="nav-link"
    href="/admin/messages"
  >
    <i
      class="fa fa-inbox"
      style="font-size: 1.5rem;"
    />
  </a>
</DocumentFragment>
`;
